db/version_edit.h

```diff
@@ -217,6 +219,9 @@ struct FileMetaData {
   // Max (newest) timestamp of keys in this file
   std::string max_timestamp;
 
+  // SST unique id
+  std::string unique_id = "";
```

Comment:
I think we should inline a 128 bit ID here. std::string is a layer of indirection that's inefficient and is flexibility we don't need. The reason the public API can emit 192-bit IDs is only for extreme (hypothetical) cases when you might need to have SST uniqueness across a very large fleet of machines, such as storing them all under a single namespace or directory in a distributed network filesystem. But for verification purposes, or even naming for a single DB directory, 128 bits is plenty.
Also, if using std::array<uint64_t, 2>, then it can be initialized with `{}` and that's different from any actual SST unique ID, internal external.

Reply:
Do you mean just use the first 128 bits of the 192 bit ID?

Reply:
Yes. https://github.com/facebook/rocksdb/blob/c4cd8e1accfd010bba9a919e234ba04505071cc1/include/rocksdb/unique_id.h#L17

I don't like the idea of tracking extra data in memory for FileMeta, but there is a convenience factor in matching the output of sst_dump. https://github.com/facebook/rocksdb/blob/c4cd8e1accfd010bba9a919e234ba04505071cc1/table/table_properties.cc#L163

I'm starting to regret supporting 192-bit IDs vs. just 128. We could still deprecate them because it's an EXPERIMENTAL API. The planned format_version=6 IDs are higher quality such that even across a large fleet, 128 bits should be plenty. https://github.com/facebook/rocksdb/pull/9058

Reply:
We rebased to #10009 and using 128bits unique id. Thanks @pdillinger for updating the public unique id length. For this feature, we switched to using internal ID.